apps/prairielearn/src/pages/instructorInstanceAdminPublishing/components/PublishingForm.tsx

```diff
@@ -0,0 +1,615 @@
+import { Temporal } from '@js-temporal/polyfill';
+import clsx from 'clsx';
+import { useEffect, useState } from 'preact/compat';
+import { Alert } from 'react-bootstrap';
+import { useForm } from 'react-hook-form';
+
+import { FriendlyDate } from '../../../components/FriendlyDate.js';
+import type { StaffCourseInstance } from '../../../lib/client/safe-db-types.js';
+import {
+  DateToPlainDateTime,
+  nowDateInTimezone,
+  plainDateTimeStringToDate,
+} from '../utils/dateUtils.js';
+
+type PublishingStatus = 'unpublished' | 'publish_scheduled' | 'published';
+
+/** Helper to compute status from dates and current time. */
+function computeStatus(
+  startDate: Date | null,
+  endDate: Date | null,
+  courseInstance: StaffCourseInstance,
+): PublishingStatus {
+  if (!startDate && !endDate) {
+    return 'unpublished';
+  }
+
+  const now = nowDateInTimezone(courseInstance.display_timezone);
```

Comment:
Why this instead of just `new Date()`? When comparing dates, timezone should never be relevant, we should just compare them.

Reply:
Fixed in 774321a74fbe818a645cbd103f4769b0007e743f.